server/src/main/java/org/apache/kafka/network/SocketServerConfigs.java

```diff
@@ -0,0 +1,146 @@
+            "If the config for the listener name is not set, the config will fallback to the generic config (i.e. <code>ssl.keystore.location</code>). " +
+            "Note that in KRaft a default mapping from the listener names defined by <code>controller.listener.names</code> to PLAINTEXT " +
+            "is assumed if no explicit mapping is provided and no other security protocol is in use.";
+
+    public static final String LISTENERS_CONFIG = "listeners";
+    public static final String LISTENERS_DEFAULT = "PLAINTEXT://:9092";
+    public static final String LISTENERS_DOC = "Listener List - Comma-separated list of URIs we will listen on and the listener names." +
+            String.format(" If the listener name is not a security protocol, <code>%s</code> must also be set.%n", LISTENER_SECURITY_PROTOCOL_MAP_CONFIG) +
+            " Listener names and port numbers must be unique unless %n" +
+            " one listener is an IPv4 address and the other listener is %n" +
+            " an IPv6 address (for the same port).%n" +
+            " Specify hostname as 0.0.0.0 to bind to all interfaces.%n" +
+            " Leave hostname empty to bind to default interface.%n" +
+            " Examples of legal listener lists:%n" +
+            " <code>PLAINTEXT://myhost:9092,SSL://:9091</code>%n" +
+            " <code>CLIENT://0.0.0.0:9092,REPLICATION://localhost:9093</code>%n" +
+            " <code>PLAINTEXT://127.0.0.1:9092,SSL://[::1]:9092</code>%n";
+
+    public static final String ADVERTISED_LISTENERS_CONFIG = "advertised.listeners";
+    public static final String ADVERTISED_LISTENERS_DOC = String.format(
+            "Listeners to publish to ZooKeeper for clients to use, if different than the <code>%s</code> config property." +
+                    " In IaaS environments, this may need to be different from the interface to which the broker binds." +
+                    " If this is not set, the value for <code>%1$1s</code> will be used." +
+                    " Unlike <code>%1$1s</code>, it is not valid to advertise the 0.0.0.0 meta-address.%n" +
+                    " Also unlike <code>%1$1s</code>, there can be duplicated ports in this property," +
+                    " so that one listener can be configured to advertise another listener's address." +
+                    " This can be useful in some cases where external load balancers are used.", LISTENERS_CONFIG);
+
+
+    public static final String CONTROL_PLANE_LISTENER_NAME_CONFIG = "control.plane.listener.name";
+    public static final String CONTROL_PLANE_LISTENER_NAME_DOC = String.format(
+            "Name of listener used for communication between controller and brokers. " +
+                    String.format("A broker will use the <code>%s</code> to locate the endpoint in %s list, to listen for connections from the controller. ", LISTENERS_CONFIG) +
```

Comment:
It lacks of `CONTROL_PLANE_LISTENER_NAME_CONFIG`

Reply:
should be fixed now

Reply:
I also removed the sub `String.format` and used the main one that wrap the whole doc